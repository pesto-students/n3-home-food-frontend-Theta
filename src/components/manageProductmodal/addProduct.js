import { UploadOutlined } from "@ant-design/icons";
import {
  Button,
  Form,
  Input,
  InputNumber,
  message,
  Modal,
  notification,
  Select,
  Upload,
} from "antd";

import axios from "axios";
import React, { useState } from "react";

import { useTranslation } from "react-i18next";
import { baseUrl } from "utils/constant";
import { getCategoryId } from "utils/helpers";

const { Option } = Select;
const openNotificationWithIcon = (type, message) => {
  notification[type]({
    message: message,
  });
};

// image
const props = {
  name: "file",
  action: "https://www.mocky.io/v2/5cc8019d300000980a055e76",
  headers: {
    authorization: "authorization-text",
  },
  onChange(info) {
    if (info.file.status !== "uploading") {
    }
    if (info.file.status === "done") {
      message.success(`${info.file.name} file uploaded successfully`);
    } else if (info.file.status === "error") {
      message.error(`${info.file.name} file upload failed.`);
    }
  },
};

const normFile = (e) => {
  if (Array.isArray(e)) {
    return e;
  }

  return e && e.fileList;
};

const CollectionCreateForm = ({ visible, onCreate, onCancel, fromFor }) => {
  const { t } = useTranslation();

  const [form] = Form.useForm();
  return (
    <Modal
      visible={visible}
      title={t("seller.product.requestForm.addProductText")}
      okText="Add"
      cancelText="Cancel"
      onCancel={onCancel}
      onOk={() => {
        form
          .validateFields()
          .then((values) => {
            form.resetFields();
            onCreate(values);
          })
          .catch((info) => {
            
          });
      }}
    >
      <Form
        form={form}
        layout="vertical"
        name="form_in_modal"
        initialValues={{
          modifier: "public",
        }}
      >
        <Form.Item
          name="productName"
          label={t("seller.product.requestForm.productName")}
          rules={[
            {
              required: true,
              message: t("seller.product.requestForm.productNameValidation"),
            },
            { min: 5, message: "Product Name must be minimum 5 characters." },
          ]}
        >
          <Input
            placeholder={t("seller.product.requestForm.productNameValidation")}
          />
        </Form.Item>
        {fromFor !== "admin" && (
          <Form.Item
            name="productCategory"
            label={t("seller.product.requestForm.productCategory")}
            rules={[
              {
                required: true,
                message: t(
                  "seller.product.requestForm.productCategoryValidation"
                ),
              },
            ]}
          >
            <Select
              mode="multiple"
              placeholder={t(
                "seller.product.requestForm.productCategoryValidation"
              )}
            >
              <Option value={getCategoryId("Breakfast")}>Breakfast</Option>
              <Option value={getCategoryId("Lunch")}>Lunch</Option>
              <Option value={getCategoryId("Snacks")}>Snacks</Option>
              <Option value={getCategoryId("Dinner")}>Dinner</Option>
            </Select>
          </Form.Item>
        )}
        <Form.Item
          name="description"
          label={t("seller.product.requestForm.description")}
          rules={[
            {
              required: true,
              message: t("seller.product.requestForm.descriptionValidation"),
            },
            {
              min: 5,
              message: "Product Description must be minimum 5 characters.",
            },
            {
              max: 20,
              message: "Product Description must be Maximun 20 characters.",
            },
          ]}
        >
          <Input
            type="textarea"
            placeholder={t("seller.product.requestForm.descriptionValidation")}
          />
        </Form.Item>

        <Form.Item
          name="productImage"
          label="Product Image"
          valuePropName="fileList"
          getValueFromEvent={normFile}
          extra=""
        >
          <Upload {...props} name="logo" maxCount={1}>
            <Button icon={<UploadOutlined />}>Click to Upload</Button>
          </Upload>
        </Form.Item>

        {fromFor === "admin" ? (
          <Form.Item
            name="maxPrice"
            label="Max Price (₹)"
            rules={[
              {
                required: true,
                message: "Please input the Max Price!",
              },
            ]}
          >
            <InputNumber min={1} placeholder="₹" />
          </Form.Item>
        ) : (
          <Form.Item
            name="maxPrice"
            label={t("seller.product.requestForm.price")}
            rules={[
              {
                required: true,
                message: t("seller.product.requestForm.priceValidation"),
              },
            ]}
          >
            <InputNumber min={1} placeholder="₹" />
          </Form.Item>
        )}
      </Form>
    </Modal>
  );
};

export const AddProductModal = (props) => {
  const [visible, setVisible] = useState(false);

  const onCreate = (values) => {
    const data = new FormData();
    data.append("image", values.productImage[0].originFileObj);
    data.append("name", values.productName);
    data.append("description", values.description);
    data.append("max_price", values.maxPrice);

    axios
      .post(
        `${baseUrl}/products/admin`,

        data
      )
      .then((result) => {
        if (result.status === 200) {
          openNotificationWithIcon("success", "Product Added");
          props.callback();
        } else {
          openNotificationWithIcon("error", "Could Not Add Product");
        }
      })
      .catch((err) => {
        console.error(err);
        openNotificationWithIcon("error", "Could Not Add Product");
      })
      .finally(() => {});
    setVisible(false);
  };

  return (
    <div>
      <div style={{ display: "flex", flexFlow: "row-reverse", margin: "10px" }}>
        <Button
          type="primary"
          onClick={() => {
            setVisible(true);
          }}
        >
          Add Product
        </Button>
      </div>

      <CollectionCreateForm
        fromFor="admin"
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};

export const AddProductSellerModal = (props) => {
  const { t } = useTranslation();

  const [visible, setVisible] = useState(false);
  const onCreate = (values) => {
    try {
      const data = new FormData();
      data.append("image", values.productImage[0].originFileObj);
      data.append("name", values.productName);
      data.append("description", values.description);
      data.append("max_price", values.maxPrice);
      data.append("category", values.productCategory);
      data.append("status", "Pending");

      axios
        .post(`${baseUrl}/products`, data)
        .then((result) => {
          if (result.status === 200) {
            openNotificationWithIcon("success", "Product Reuested");
            props.callback();
          } else {
            openNotificationWithIcon("error", "Could Not Add Product");
          }
        })
        .catch((err) => {
          console.error(err);
          openNotificationWithIcon("error", "Could Not Add Product");
        })
        .finally(() => {});
      setVisible(false);
    } catch (e) {
      openNotificationWithIcon("error", "Something went wrong");
    }
  };

  return (
    <div>
      <div style={{ display: "flex", flexFlow: "row-reverse", margin: "10px" }}>
        <Button
          type="primary"
          onClick={() => {
            setVisible(true);
          }}
        >
          {t("seller.product.requestProductButton")}
        </Button>
      </div>

      <CollectionCreateForm
        fromFor="seller"
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};
