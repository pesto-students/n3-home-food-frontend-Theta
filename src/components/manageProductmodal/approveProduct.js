import { Button, Form, InputNumber, Modal, notification } from "antd";
import "antd/dist/antd.css";
import axios from "axios";
import React, { useState } from "react";

import { baseUrl } from "utils/constant";

const openNotificationWithIcon = (type, message) => {
  notification[type]({
    message: message,
  });
};

const CollectionCreateForm = ({ visible, onCreate, onCancel }) => {
  const [form] = Form.useForm();
  return (
    <Modal
      visible={visible}
      title="Approve Product"
      okText="Approve"
      cancelText="Cancel"
      onCancel={onCancel}
      onOk={() => {
        form
          .validateFields()
          .then((values) => {
            form.resetFields();
            onCreate(values);
          })
          .catch((info) => {});
      }}
    >
      <Form
        form={form}
        layout="vertical"
        name="form_in_modal"
        initialValues={{
          modifier: "public",
        }}
      >
        <Form.Item
          name="maxPrice"
          label="Max Price"
          rules={[
            {
              required: true,
              message: "Please input the Max Price!",
            },
          ]}
        >
          <InputNumber min={1} />
        </Form.Item>
      </Form>
    </Modal>
  );
};

export const AppoveProductModal = (props) => {
  const [visible, setVisible] = useState(false);

  const onCreate = (values) => {
    axios
      .put(`${baseUrl}/products/product-approval/${props.productId}`, {
        max_price: values.maxPrice,
      })
      .then((result) => {
        if (result.status === 200) {
          openNotificationWithIcon("success", "Product Approved");
          props.callback();
        } else {
          openNotificationWithIcon("error", "Could Not Approve Product");
        }
      })
      .catch((err) => {
        console.error(err);
        openNotificationWithIcon("error", "Could Not Add Product");
      })
      .finally(() => {});
    setVisible(false);
  };

  return (
    <div>
      <div style={{ display: "flex", flexFlow: "row-reverse", margin: "10px" }}>
        <Button
          type="primary"
          onClick={() => {
            setVisible(true);
          }}
        >
          Approve
        </Button>
      </div>

      <CollectionCreateForm
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};
