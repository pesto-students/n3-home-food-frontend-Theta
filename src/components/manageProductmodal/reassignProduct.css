
.ant-upload-text-icon  .anticon-loading{
display: none !important
}