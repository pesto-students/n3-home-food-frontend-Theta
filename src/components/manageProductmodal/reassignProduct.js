import { Button, Form, Modal, notification, Select } from "antd";

import React, { useEffect, useState } from "react";
import { reassignProduct, getApproveProduct } from "../utils/api";

let { Option } = Select;
const openNotificationWithIcon = (type, message) => {
  notification[type]({
    message: message,
  });
};

const CollectionCreateForm = ({ visible, onCreate, onCancel }) => {
  const [form] = Form.useForm();

  const [products, setproducts] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const getProducts = async () => {
      const response = await getApproveProduct().catch(() => {
        setIsLoading(false);
      });
      setproducts(response.data);
    };
    getProducts();
  }, []);

  return (
    <Modal
      visible={visible}
      title="Reassign Product"
      okText="Reassign"
      cancelText="Cancel"
      onCancel={onCancel}
      onOk={() => {
        form
          .validateFields()
          .then((values) => {
            form.resetFields();
            onCreate(values);
          })
          .catch((info) => {});
      }}
    >
      {isLoading}
      <Form
        form={form}
        layout="vertical"
        name="form_in_modal"
        initialValues={{
          modifier: "public",
        }}
      >
        <Form.Item
          name="existingProductId"
          label="Choose Product"
          rules={[
            {
              required: true,
              message: "Please select the Product!",
            },
          ]}
        >
          <Select placeholder="Please select a Product">
            {products.map((product, i) => (
              <Option value={product.id}>
                <span>{product.name}</span>
                {/* <span className="cost">
                  <p className="max-amount">Max Amount</p> &nbsp; â‚¹
                  {product.max_price}
                </span> */}
              </Option>
            ))}
          </Select>
        </Form.Item>
      </Form>
    </Modal>
  );
};

export const ReassignProduct = (props) => {
  const [visible, setVisible] = useState(false);

  const onCreate = async (values) => {
    const response = await reassignProduct(props.productId, {
      existingProductId: values.existingProductId,
    }).catch(() => {
      openNotificationWithIcon("error", "Could Not Reassigned Product");
    });
    if (response.status === 200) {
      openNotificationWithIcon("success", "Product Reassigned");
      props.callback();
    } else {
      openNotificationWithIcon("error", "Could Not Reassign Product ");
    }

    setVisible(false);
  };

  return (
    <div>
      <div style={{ display: "flex", flexFlow: "row-reverse", margin: "10px" }}>
        <Button
          type="primary"
          onClick={() => {
            setVisible(true);
          }}
        >
          Reassign
        </Button>
      </div>

      <CollectionCreateForm
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};
