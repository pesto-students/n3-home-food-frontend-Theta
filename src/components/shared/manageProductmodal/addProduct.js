import { UploadOutlined } from "@ant-design/icons";
import { Button, Form, Input, InputNumber, message, Modal, notification, Select, Upload } from "antd";
import "antd/dist/antd.css";
import axios from 'axios';
import React, { useState } from "react";
const { Option } = Select;



const openNotificationWithIcon = (type,message) => {
  notification[type]({
    message: message
  });
};

// image
const props = {
  name: "file",
  action: "https://www.mocky.io/v2/5cc8019d300000980a055e76",
  headers: {
    authorization: "authorization-text",
  },
  onChange(info) {
    if (info.file.status !== "uploading") {
      console.log(info.file, info.fileList);

     
    }
    if (info.file.status === "done") {
      message.success(`${info.file.name} file uploaded successfully`);
    } else if (info.file.status === "error") {
      message.error(`${info.file.name} file upload failed.`);
    }
  },
};

const normFile = (e) => {
  console.log("Upload event:", e);

  if (Array.isArray(e)) {
    return e;
  }

  return e && e.fileList;
};

const CollectionCreateForm = ({ visible, onCreate, onCancel,fromFor }) => {
  const [form] = Form.useForm();
  return (
    <Modal
      visible={visible}
      title="Add Product"
      okText="Add"
      cancelText="Cancel"
      onCancel={onCancel}
      onOk={() => {
        form
          .validateFields()
          .then((values) => {
            form.resetFields();
            onCreate(values);
          })
          .catch((info) => {
            console.log("Validate Failed:", info);
          });
      }}
    >
      <Form
        form={form}
        layout="vertical"
        name="form_in_modal"
        initialValues={{
          modifier: "public",
        }}
      >
        <Form.Item
          name="productName"
          label="Product Name"
          rules={[
            {
              required: true,
              message: "Please input the Product Name!",
            },
          ]}
        >
          <Input />
        </Form.Item>

        <Form.Item
          name="productCategory"
          label="Product Category"
          rules={[
            {
              required: true,
              message: "Please input the Product Category!",
            },
          ]}
        >
             <Select placeholder="Please select a Category">
          <Option value="60c906ce35453e14cd3f4ee3">Breakfast</Option>
          <Option value="Lunch">Lunch</Option>
          <Option value="Snacks">Snacks</Option>
          <Option value="Dinner">Dinner</Option>
        </Select>
        </Form.Item>

        <Form.Item name="description" label="Description"
          rules={[
            {
              required: true,
              message: "Please input the Product Description!",
            },
          ]}>
          <Input type="textarea" />
        </Form.Item>

        <Form.Item
          name="productImage"
          label="Product Image"
          valuePropName="fileList"
          getValueFromEvent={normFile}
          extra=""
        >
    
          <Upload {...props} name="logo" maxCount={1} >
            <Button  icon={<UploadOutlined />}>Click to Upload</Button>
          </Upload>
        </Form.Item>
       
       {fromFor === 'admin' ?
        <Form.Item
          name="maxPrice"
          label="Max Price"
          rules={[
            {
              required: true,
              message: "Please input the Max Price!",
            },
          ]}
        >
          <InputNumber min={1}  />
        </Form.Item>
        :
        <Form.Item
        name="Price"
        label="Price"
        rules={[
          {
            required: true,
            message: "Please Enter Price!",
          },
        ]}
      >
        <InputNumber min={1}  />
      </Form.Item>

}

      
      </Form>
    </Modal>
  );
};

export const AddProductModal = (props) => {
  const [visible, setVisible] = useState(false);

    const onCreate = (values) => {
    
    const data = new FormData()
    data.append('image',values.productImage[0].originFileObj)
    data.append('name',values.productName)
    data.append('description',values.description)
    data.append('max_price',values.maxPrice)
    data.append('category',values.productCategory)

    console.log('file,',data)
    axios
    .post("http://localhost:8080/api/v1/products",

    data)
    .then((result) => {
      if(result.status === 200){
        openNotificationWithIcon('success','Product Added')
      }
      else{
        openNotificationWithIcon('error','Could Not Add Product')

      }
      
    })
    .catch((err) => {
      console.error(err)
      openNotificationWithIcon('error','Could Not Add Product')
    })
    .finally(() => {

    });
    setVisible(false);
  };

  return (
    <div>
      <div style={{ display: "flex", flexFlow: "row-reverse", margin: "10px" }}>
        <Button
          type="primary"
          onClick={() => {
            setVisible(true);
          }}
        >
          Add Product
        </Button>
      </div>

      <CollectionCreateForm
        fromFor='admin'
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};



export const AddProductSellerModal = (props) => {
  const [visible, setVisible] = useState(false);

  const onCreate = (values) => {
    
    const data = new FormData()
    data.append('image',values.productImage[0].originFileObj)
    data.append('name',values.productName)
    data.append('description',values.description)
    data.append('price',values.maxPrice)
    data.append('category',values.productCategory)

    axios
    .post("http://localhost:8080/api/v1/products",

    data)
    .then((result) => {
      if(result.status === 200){
        openNotificationWithIcon('success','Product Reuested')
      }
      else{
        openNotificationWithIcon('error','Could Not Add Product')

      }
      
    })
    .catch((err) => {
      console.error(err)
      openNotificationWithIcon('error','Could Not Add Product')
    })
    .finally(() => {

    });
    setVisible(false);
  };

  return (
    <div>
      <div style={{ display: "flex", flexFlow: "row-reverse", margin: "10px" }}>
        <Button
          type="primary"
          onClick={() => {
            setVisible(true);
          }}
        >
          Reuest Product
        </Button>
      </div>

      <CollectionCreateForm
        fromFor='seller'
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};
