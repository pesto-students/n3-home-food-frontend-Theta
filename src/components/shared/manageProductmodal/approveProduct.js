import { Button, Form, InputNumber, Modal, notification } from "antd";
import "antd/dist/antd.css";
import axios from 'axios';
import React, { useState } from "react";



const openNotificationWithIcon = (type,message) => {
    notification[type]({
      message: message
    });
  };
  
const CollectionCreateForm = ({ visible, onCreate, onCancel }) => {
  const [form] = Form.useForm();
  return (
    <Modal
      visible={visible}
      title="Approve Product"
      okText="Approve"
      cancelText="Cancel"
      onCancel={onCancel}
      onOk={() => {
        form
          .validateFields()
          .then((values) => {
            form.resetFields();
            onCreate(values);
          })
          .catch((info) => {
            console.log("Validate Failed:", info);
          });
      }}
    >
      <Form
        form={form}
        layout="vertical"
        name="form_in_modal"
        initialValues={{
          modifier: "public",
        }}
      >

        <Form.Item
          name="maxPrice"
          label="Max Price"
          rules={[
            {
              required: true,
              message: "Please input the Max Price!",
            },
          ]}
        >
          <InputNumber min={1}  />
        </Form.Item>

      
      </Form>
    </Modal>
  );
};

export const AppoveProductModal = (props) => {
  const [visible, setVisible] = useState(false);

  const onCreate = (values) => {
    console.log("Received values of form: ", values);

    axios
    .put(`http://localhost:8080/api/v1/products/product-approval/${props.productId}`,{'max_price':values.maxPrice})
    .then((result) => {
      if(result.status === 200){
        openNotificationWithIcon('success','Product Approved')
      }
      else{
        openNotificationWithIcon('error','Could Not Approve Product')

      }
      
    })
    .catch((err) => {
      console.error(err)
      openNotificationWithIcon('error','Could Not Add Product')
    })
    .finally(() => {

    });
    setVisible(false);
  };

  return (
    <div>
      <div style={{ display: "flex", flexFlow: "row-reverse", margin: "10px" }}>
        <Button
          type="primary"
          onClick={() => {
            setVisible(true);
          }}
        >
          Approve
        </Button>
      </div>

      <CollectionCreateForm
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};
