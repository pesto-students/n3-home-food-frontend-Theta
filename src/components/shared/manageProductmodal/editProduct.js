import { UploadOutlined } from "@ant-design/icons";
import {
  Button,
  Form,
  Input,
  InputNumber,
  message,
  Modal,
  notification,
  Upload
} from "antd";
import "antd/dist/antd.css";
import axios from "axios";
import React, { useState } from "react";
import { baseUrl } from "../../../utils/constant";

const openNotificationWithIcon = (type, message) => {
  notification[type]({
    message: message,
  });
};

// image
const props = {
  name: "file",
  action: "https://www.mocky.io/v2/5cc8019d300000980a055e76",
  headers: {
    authorization: "authorization-text",
  },
  onChange(info) {
    if (info.file.status !== "uploading") {
      console.log(info.file, info.fileList);
    }
    if (info.file.status === "done") {
      message.success(`${info.file.name} file uploaded successfully`);
    } else if (info.file.status === "error") {
      message.error(`${info.file.name} file upload failed.`);
    }
  },
};

const normFile = (e) => {
  console.log("Upload event:", e);

  if (Array.isArray(e)) {
    return e;
  }

  return e && e.fileList;
};

const CollectionCreateForm = ({
  product,
  visible,
  onCreate,
  onCancel,
  fromFor,
}) => {
  const [form] = Form.useForm();
  const fileList = [
    {
      uid: "-1",
      status: "done",
      url: product.image,
      thumbUrl: product.image,
    },
  ];
  return (
    <Modal
      visible={visible}
      title="Edit Product"
      okText="Edit"
      cancelText="Cancel"
      onCancel={onCancel}
      onOk={() => {
        form
          .validateFields()
          .then((values) => {
            form.resetFields();
            onCreate(values);
          })
          .catch((info) => {
            console.log("Validate Failed:", info);
          });
      }}
    >
      <Form
        form={form}
        layout="vertical"
        name="form_in_modal"
        initialValues={{
          productName: product.name,
          description: product.description,
          maxPrice: product.max_price,
        }}
      >
        <Form.Item
          name="productName"
          label="Product Name"
          rules={[
            { min: 5, message: "Product Name must be minimum 5 characters." },
          ]}
        >
          <Input />
        </Form.Item>

        <Form.Item
          name="description"
          label="Description"
          rules={[
            {
              min: 5,
              message: "Product Description must be minimum 5 characters.",
            },
            {
              max: 20,
              message: "Product Description must be Maximun 20 characters.",
            },
          ]}
        >
          <Input type="textarea" />
        </Form.Item>

        <Form.Item
          name="productImage"
          label="Product Image"
          valuePropName="fileList"
          getValueFromEvent={normFile}
          extra=""
        >
          <Upload
            {...props}
            name="logo"
            maxCount={1}
            listType="picture"
            defaultFileList={[...fileList]}
          >
            <Button icon={<UploadOutlined />}>Click to Upload</Button>
          </Upload>
        </Form.Item>

        {fromFor === "admin" ? (
          <Form.Item name="maxPrice" label="Max Price (₹)">
            <InputNumber min={1} placeholder="₹" />
          </Form.Item>
        ) : (
          <Form.Item
            name="maxPrice"
            label="Price (₹)"
            rules={[
              {
                required: true,
                message: "Please Enter Price!",
              },
            ]}
          >
            <InputNumber min={1} placeholder="₹" />
          </Form.Item>
        )}
      </Form>
    </Modal>
  );
};

export const EditProductModal = (props) => {
  const [visible, setVisible] = useState(false);

  const onCreate = (values) => {
    const data = new FormData();

    if (typeof values.productImage === "undefined") {
      data.append("image", props.product.image);
    } else {
      data.append("image", values.productImage[0].originFileObj);
    }

    data.append("name", values.productName);
    data.append("description", values.description);
    data.append("max_price", values.maxPrice);
    data.append("category", values.productCategory);

    console.log("file,", data);
    axios
      .put(`${baseUrl}/products/${props.product.id}`, data)
      .then((result) => {
        if (result.status === 200) {
          openNotificationWithIcon("success", "Product Edited");
          props.callback();
        } else {
          openNotificationWithIcon("error", "Could Not Edited Product");
        }
      })
      .catch((err) => {
        console.error(err);
      })
      .finally(() => {});
    setVisible(false);
  };

  return (
    <div>
      <div
        onClick={() => {
          setVisible(true);
        }}
      >
        Edit
      </div>
      <CollectionCreateForm
        product={props.product}
        fromFor="admin"
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};
