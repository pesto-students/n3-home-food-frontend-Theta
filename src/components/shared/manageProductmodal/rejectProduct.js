import { Button, Form, Input, Modal, notification } from "antd";
import "antd/dist/antd.css";
import axios from 'axios';
import React, { useState } from "react";

import { baseUrl } from "../../../utils/constant";


const openNotificationWithIcon = (type,message) => {
    notification[type]({
      message: message
    });
  };
  
const CollectionCreateForm = ({ visible, onCreate, onCancel }) => {
  const [form] = Form.useForm();
  return (
    <Modal
      visible={visible}
      title="Reject Product"
      okText="Reject"
      cancelText="Cancel"
      onCancel={onCancel}
      onOk={() => {
        form
          .validateFields()
          .then((values) => {
            form.resetFields();
            onCreate(values);
          })
          .catch((info) => {
            console.log("Validate Failed:", info);
          });
      }}
    >
      <Form
        form={form}
        layout="vertical"
        name="form_in_modal"
        initialValues={{
          modifier: "public",
        }}
      >

<Form.Item
          name="rejectReason"
          label="Reject Reason"
          rules={[
            {
              required: true,
              message: "Please input the Rejection Reason!",
            },
            { min: 5, message: "Rejection Reason must be minimum 5 characters." },
          ]}
        >
          <Input />
        </Form.Item>

      
      </Form>
    </Modal>
  );
};

export const RejectProductModal = (props) => {
  const [visible, setVisible] = useState(false);

  const onCreate = (values) => {
    console.log("Received values of form: ", values);

    axios
    .put(`${baseUrl}/products/product-rejection/${props.productId}`,{'rejection_reason':values.rejectReason})
    .then((result) => {
      if(result.status === 200){
        openNotificationWithIcon('success','Product Rejected')
        props.callback()
      }
      else{
        openNotificationWithIcon('error','Could Not Reject Product ')

      }
      
    })
    .catch((err) => {
      console.error(err)
      openNotificationWithIcon('error','Could Not Reject Product')
    })
    .finally(() => {

    });
    setVisible(false);
  };

  return (
    <div>
      <div style={{ display: "flex", flexFlow: "row-reverse", margin: "10px" }}>
        <Button
          type="primary"
          onClick={() => {
            setVisible(true);
          }}
        >
          Reject
        </Button>
      </div>

      <CollectionCreateForm
        visible={visible}
        onCreate={onCreate}
        onCancel={() => {
          setVisible(false);
        }}
      />
    </div>
  );
};
