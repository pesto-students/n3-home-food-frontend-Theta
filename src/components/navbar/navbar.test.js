import * as React from "react";
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import { RejectProductModal } from "components/manageProductmodal/rejectProduct";

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Navbar component", () => {
  let wrapper;
  test("render without error", () => {
    wrapper = shallow(<RejectProductModal/>);
    expect(wrapper.length).toBe(1);
  });

  test('should render correctly in "debug" mode', () => {
    const component = shallow(<RejectProductModal />);

    expect(component).toMatchSnapshot();
  });
});