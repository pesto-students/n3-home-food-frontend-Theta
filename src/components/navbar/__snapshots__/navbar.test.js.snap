// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
